x-pack/plugins/ml/public/application/routing/routes/new_job/index_or_search.tsx

```diff
@@ -174,6 +174,26 @@ export const changePointDetectionIndexOrSearchRouteFactory = (
   breadcrumbs: getChangePointDetectionBreadcrumbs(navigateToPath, basePath),
 });
 
+export const dataComparisonIndexOrSearchRouteFactory = (
+  navigateToPath: NavigateToPath,
+  basePath: string
+): MlRoute => ({
+  id: 'data_view_data_compare',
+  path: createPath('data_compare_index_select'),
```

Comment:
Suggest to rename `data_compare_index_select` to `data_comparison_index_select`.